src/python.rs

```diff
@@ -61,7 +61,13 @@ impl<'p> Python<'p> {
     /// I.e., `Python<'static>` is always *really* unsafe.
     #[inline]
     pub unsafe fn assume_gil_acquired() -> Python<'p> {
-        Python(PhantomData)
+        let count = gil::get_gil_count();
+        debug_assert!(count > 0);
+        Python(count, PhantomData)
```

Comment:
I think this isn't quite right, because you could have this:

```rust
let gil = Python::acquire_gil();
let py = gil.python();

let list = PyList::new(py, &[1, 2, 3]);

let gil2 = Python::acquire_gil();
// Because of assume_gil_acquired(), it is put into gil2's pool.
// But the Rust compiler thinks `item` has the same lifetime as `list`, i.e. `gil1` 
let item = list.get_item(0);          

// Rust thinks this is fine, but item is now dangling.
drop(gil2)
```

I think the only correct way for functions like `list.get_item()` to work is that `item` has to go into the same `GILPool` as `list`, because that's the lifetime that rust sees.

To do this, I think all py native types would have to store `Python`. Maybe it's not so bad, but it doesn't feel great...

Reply:
Ah, you're right. Thank you